### PR TITLE
Update IAST redaction algorithm

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Updates the redaction algorithm to be more lenient

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/model/json/VulnerabilityEncodingTest.groovy [hunk 1]
```diff
@@ -19,6 +19,22 @@ class VulnerabilityEncodingTest extends DDSpecification {
...
+    final value = new VulnerabilityBatch()
+    value.add(null)
+
+    when:
+    final result = VulnerabilityEncoding.toJson(value)
+
+    then:
+    JSONAssert.assertEquals('''{
+      "vulnerabilities": [
+        null
```
Comment (smola) [inline dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/model/json/VulnerabilityEncodingTest.groovy:33]: Do we really want to do this? We should probably skip the element altogether?

Comment (manuel-alvarez-alvarez) [reply dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/model/json/VulnerabilityEncodingTest.groovy:33]: Yep it's quite weird. It was like this before the PR but jacoco forced me to add the tests due to coverage, while doing the tests I thought about changing it, so I'll skip the vulnerability if it's null.

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/model/json/EvidenceAdapter.java [hunk 2]
```diff
@@ -15,6 +16,8 @@
 import com.squareup.moshi.Moshi;
 import datadog.trace.api.Config;
 import java.io.IOException;
+import java.util.*;
```
Comment (smola) [inline dd-java-agent/agent-iast/src/main/java/com/datadog/iast/model/json/EvidenceAdapter.java:19]: We generally avoid wildcard imports: https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#intellij-idea

**Review by smola — APPROVED** [review]

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/model/json/EvidenceAdapter.java (hunk 2 above)
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/agent-iast/src/main/java/com/datadog/iast/model/json/EvidenceAdapter.java]: Yeah, I copied from another project that did not have the settings and IntelliJ did not change it :cry: 

Updated, thanks!
